### ISSUE TITLE
Add and update existing tests for challenge submission API

### ISSUE THREAD

**Issue by Sanji515** [Cloud-CV/EvalAI#2443]
Reference: #2409 

- Some tests were removed in the above mentioned PR but we need those tests with some updation when we upload a file for challenge submission.

- Also need to add tests when we make submission using URL.

**Comment by coldk91ght** [comment]
@Sanji515 Can I take up this issue?

**Comment by Sanji515** [comment]
Sure @coldk91ght

**Comment by coldk91ght** [comment]
@Sanji515 Ohkay. On it. 👍

**Comment by coldk91ght** [comment]
@Sanji515 Shall I make a separate class for submission post tests?  Also, can you please lead me to the input format for url uploads?

**Comment by r4zorgeek** [comment]
@Sanji515 Can I take this issue?

**Comment by coldk91ght** [comment]
I'm sorry for the delay. A close relative of mine had passed away and so I couldn't focus on the task. I have put some new tests. I need some help with the successful url tests and to make a sample input file url for testing. These are the updated tests for now: https://github.com/coldk91ght/EvalAI/commit/38480d99cedd377194fcf8d83e89add0b77750ff

**Comment by pushkalkatara** [comment]
Hi @nikochiko These tests might help you out -  [test_submission_worker.py](https://github.com/Cloud-CV/EvalAI/blob/ca0351dda61be800b2ee50a2b9d4f41b913e8ea9/tests/integration/worker/test_submission_worker.py#L231)

**Comment by RishabhJain2018** [comment]
Thanks, @pushkalkatara for the pointer.

**Comment by RishabhJain2018** [comment]
@nikochiko Please open up the PR for it.

**Comment by nikochiko** [comment]
@pushkalkatara Thanks for the help.
Opening up PR now. Will add changes to it.

**Comment by pushkalkatara** [comment]
Hi @nikochiko . I think these two tests are also required which were removed - [status, file not submitted](https://github.com/Cloud-CV/EvalAI/pull/2409/files#diff-c6268b43fc26de9e08ed355975be069bL424) and [wrong form encoding](https://github.com/Cloud-CV/EvalAI/pull/2409/files#diff-c6268b43fc26de9e08ed355975be069bL445). @Sanji515 Can you comment on this?

**Comment by Sanji515** [comment]
> Hi @nikochiko . I think these two tests are also required which were removed - [status, file not submitted](https://github.com/Cloud-CV/EvalAI/pull/2409/files#diff-c6268b43fc26de9e08ed355975be069bL424) and [wrong form encoding](https://github.com/Cloud-CV/EvalAI/pull/2409/files#diff-c6268b43fc26de9e08ed355975be069bL445). @Sanji515 Can you comment on this?

Yes, @nikochiko Those tests need to be added, they may need some updation. Thanks @pushkalkatara for pointing the missing tests

**Comment by nikochiko** [comment]
@pushkalkatara , thanks for pointing. @Sanji515 , I'm working on it.

**Comment by nikochiko** [comment]
@pushkalkatara @Sanji515 I need some help with these new tests. I'm struggling with adding these tests without `KeyError` in the Travis CI build. Any pointers please?

**Comment by nikochiko** [comment]
@Sanji515 Done. Please review.
